packages/grainIntegration-merkle/contracts/MerkleRedeem.sol

```diff
@@ -0,0 +1,213 @@
+     * Will return false for a non-existent distribution.
+     *
+     * The `_end` value is included in the returned array
+     */
+    function claimStatus(
+        address _contributor,
+        uint _begin,
+        uint _end
+    )
+        external
+        view
+        returns (bool[] memory)
+    {
+        uint size = 1 + _end - _begin;
+        bool[] memory arr = new bool[](size);
+        for(uint i = 0; i < size; i++) {
+            arr[i] = claimed[_begin + i][_contributor];
+        }
+        return arr;
+    }
+
+    /**
+     * Get the bytes32 hash merkle root for a slice of all valid
+     * distributions. This will return bytes32(0) if a distribution is
+     * paused or non-existent. Non-existent distributions have a `claimTime`
+     * of zero.
+     *
+     * The `_end` value is included in the returned array
+     */
+    function merkleRoots(
+        uint _begin,
+        uint _end
+    )
+        external
+        view
+        returns (bytes32[] memory)
+    {
+        uint size = 1 + _end - _begin;
+        bytes32[] memory arr = new bytes32[](size);
+        for(uint i = 0; i < size; i++) {
+            arr[i] = distributionMerkleRoots[_begin + i];
+        }
+        return arr;
+    }
+
+    /**
+     *  Check the validity of a claim attempt
+     */
+    function verifyClaim(
+        address _contributor,
+        uint _distribution,
+        uint _claimedBalance,
+        bytes32[] memory _merkleProof
+    )
+        public
+        view
+        returns (bool valid)
+    {
+        bytes32 leaf = keccak256(abi.encodePacked(_contributor, _claimedBalance));
+        return MerkleProof.verify(_merkleProof, distributionMerkleRoots[_distribution], leaf);
+    }
+
+    /**
+     * Publish a merkleRoot that contributors can validate against to claim
+     * funds. Currently the _claimTime can be set to any time past or present,
+     * but this may not be the case when ultimately released. Distributions
+     * might need to observe some minimum delay period before claims
+     * can successfully execute against it.
+     */
+    function seedDistribution(
+        uint _distribution,
+        bytes32 _merkleRoot,
+        uint64 _claimTime
+    )
+        external
+    {
+        require(hasRole(SEEDER_ROLE, msg.sender), "MerkleRedeem: Must have SEEDER_ROLE to post allocation roots");
+        require(distributionMerkleRoots[_distribution] == bytes32(0), "MerkleRedeem: cannot rewrite merkle root");
+        require(block.timestamp + minDelay <= _claimTime, "MerkleRedeem: Delay is too short");
+        distributionMerkleRoots[_distribution] = _merkleRoot;
+        claimTimes[_distribution] = _claimTime;
+        emit DistributionPublished(_distribution, _merkleRoot);
+    }
+
+    /**
+     * Pause a Distribution indefinitely by resetting the merkle root to zero
+     * The distribution can be re-seeded with a new merkle root.
+     */
+    function pauseDistribution(uint _distribution) external {
```

Comment:
should be `deleteDistribution`

Reply:
This is because we are setting the bytes to 0, even though we can effectively "undelete/unpause"